packages/eslint-plugin/src/rules/require-array-sort-compare.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @fileoverview Enforce giving `compare` argument to `Array#sort`
+ * @author Toru Nagashima <https://github.com/mysticatea>
+ */
+
+import * as ts from 'typescript';
+import { TSESTree } from '@typescript-eslint/typescript-estree';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'require-array-sort-compare',
+  defaultOptions: [],
+
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Enforce giving `compare` argument to `Array#sort`',
+      category: 'Best Practices',
+      recommended: false
+    },
+    messages: {
+      requireCompare: "Require 'compare' argument."
+    },
+    schema: []
+  },
+
+  create(context) {
+    const service = util.getParserServices(context);
+    const checker = service.program.getTypeChecker();
+
+    return {
+      CallExpression(node: TSESTree.CallExpression) {
```

Comment:
Could remove a number of `if` checks below if you leverage selectors
```suggestion
      'CallExpression[arguments.length = 0] > MemberExpression[computed = false]'(node: TSESTree.MemberExpression) {
```